File: src/main/java/kr/hailor/hailor/controller/HealthController.java
package kr.hailor.hailor.controller;

import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HealthController {

    @GetMapping("/health")
    HealthResponse health() {
        return new HealthResponse(HttpServletResponse.SC_OK);
    }

    @GetMapping("/ready")
    void ready() {
    }
}


record HealthResponse(int status) {
}

File: src/main/java/kr/hailor/hailor/controller/HealthController.kt
package kr.hailor.hailor.controller

import io.swagger.v3.oas.annotations.Hidden
import jakarta.servlet.http.HttpServletResponse
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@Hidden
@RestController
class HealthController {
    @GetMapping("/health")
    fun health(): HealthResponse = HealthResponse(HttpServletResponse.SC_OK)

    @GetMapping("/ready")
    fun ready() {
    }
}

@JvmRecord
data class HealthResponse(
    val status: Int,
)
